source/frontend/src/features/documents/ComposedDocument.ts

```diff
@@ -1,84 +1,99 @@
-  parentId: number | undefined;
+  relationshipType: ApiGen_CodeTypes_DocumentRelationType | null = null;
+  parentId: string | undefined;
 
-  public static fromApi(relationship: Api_DocumentRelationship): DocumentRow {
+  public static fromApi(relationship: ApiGen_Concepts_DocumentRelationship): DocumentRow {
     const row: DocumentRow = new DocumentRow();
     row.id = relationship.document?.id;
     row.documentType = relationship.document?.documentType ?? undefined;
     row.mayanDocumentId = relationship.document?.mayanDocumentId ?? undefined;
     row.statusTypeCode = relationship.document?.statusTypeCode ?? undefined;
     row.fileName = relationship.document?.fileName ?? undefined;
     row.appCreateTimestamp = relationship.document?.appCreateTimestamp;
-    row.appCreateUserid = relationship.document?.appCreateUserid;
+    row.appCreateUserid = relationship.document?.appCreateUserid ?? undefined;
 
     row.relationshipId = relationship.id;
-    row.relationshipType = relationship.relationshipType ?? undefined;
+    row.relationshipType = relationship.relationshipType ?? null;
     row.parentId = relationship.parentId ?? undefined;
     return row;
   }
 
-  public static fromApiDocument(document: Api_Document): DocumentRow {
+  public static fromApiDocument(document: ApiGen_Concepts_Document): DocumentRow {
     const row: DocumentRow = new DocumentRow();
     row.id = document?.id;
     row.documentType = document?.documentType ?? undefined;
     row.mayanDocumentId = document?.mayanDocumentId ?? undefined;
     row.statusTypeCode = document?.statusTypeCode ?? undefined;
     row.fileName = document?.fileName ?? undefined;
     row.appCreateTimestamp = document?.appCreateTimestamp;
-    row.appCreateUserid = document?.appCreateUserid;
+    row.appCreateUserid = document?.appCreateUserid ?? undefined;
 
     row.relationshipId = undefined;
-    row.relationshipType = undefined;
+    row.relationshipType = null;
     row.parentId = undefined;
     return row;
   }
 
-  public static toApi(document: DocumentRow): Api_DocumentRelationship {
+  public static toApi(document: DocumentRow): ApiGen_Concepts_DocumentRelationship {
+    if (document.relationshipType === null) {
+      throw new Error('Invalid document relationship type');
+    }
+
     return {
-      id: document.relationshipId,
-      relationshipType: document.relationshipType ?? null,
+      id: document.relationshipId || 0,
+      relationshipType: document.relationshipType,
       parentId: document.parentId ?? null,
       document: {
-        id: document.id,
-        mayanDocumentId: document.mayanDocumentId ?? null,
+        id: document.id || 0,
+        mayanDocumentId: document.mayanDocumentId ?? 0,
         documentType: document.documentType ?? null,
         statusTypeCode: document.statusTypeCode ?? null,
         fileName: document.fileName ?? null,
+        rowVersion: 1,
+
+        appCreateTimestamp: '',
```

Comment:
would it make more sense to change the backend models so these are nullable?

Reply:
I also wonder if we could somehow make incorporating these common auditing elements more readable? kind of like how we do it on the backend where an underlying mapper handles these audit fields so we don't have to repeat them everywhere.

Reply:
tbh, I dont think the frontend needs those at all. Probably only some of those are actually needed. i.e create date time, user name